test/python/common.py

```diff
@@ -223,6 +224,41 @@ def _(*args, **kwargs):
     return _
 
 
+def is_cpp_simulator_available():
+    """
+    Check if executable for C++ simulator is available in the expected
+    location.
+
+    Returns:
+        bool: True if simulator executable is available
+    """
+    try:
+        QasmSimulatorCpp()
+    except FileNotFoundError:
+        return False
+    return True
+
+
+def requires_cpp_simulator(func):
```

Comment:
Let's define this decorator in terms of the [`unittest.skipIf`](https://docs.python.org/3.7/library/unittest.html#unittest.skipIf) so we can use as a class or function decorator, indistinctly:

```python
def requires_cpp_simulator(test_item):
    reason = 'C++ simulator not found, skipping test'
    return unittest.skipIf(is_cpp_simulator_available(), reason)(test_item)
```